### PR TITLE
Merge Properties and PropertiesI, switch to string_view

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR merges Properties and PropertiesI and replaces "const string&" by string_view in the Properties API.

### CODE REVIEW

File: cpp/include/Ice/Communicator.h [hunk 1]
```diff
@@ -13,11 +13,14 @@
 #include "Initialize.h"
 #include "PluginF.h"
 #include "Properties.h"
-#include "ProxyF.h"
+#include "Proxy.h"
```
Comment (bernardnormier) [inline cpp/include/Ice/Communicator.h:16]: My cleanup of the headers included by Properties.h revealed a bug in this include list.

File: cpp/include/Ice/Initialize.h [hunk 2]
```diff
@@ -117,7 +117,7 @@ ICE_API PropertiesPtr createProperties();
  * @return A new property set initialized with the property settings
  * that were removed from the argument vector.
  */
-ICE_API PropertiesPtr createProperties(StringSeq& seq, const PropertiesPtr& defaults = 0);
+ICE_API PropertiesPtr createProperties(StringSeq& seq, const PropertiesPtr& defaults = nullptr);
```
Comment (bernardnormier) [inline cpp/include/Ice/Initialize.h:120]: I am wonder if we should just remove these functions that are now equivalent to a call to `make_shared<Properties>(...)`

File: cpp/src/Ice/Properties.cpp [hunk 3]
```diff
@@ -2,8 +2,729 @@
...
+using namespace Ice;
+using namespace IceInternal;
+
+Ice::Properties::Properties(const Properties& p)
+{
+    lock_guard lock(const_cast<Properties&>(p)._mutex);
+    _properties = p._properties;
+}
+
+Ice::Properties::Properties(StringSeq& args, const PropertiesPtr& defaults)
```
Comment (bernardnormier) [inline cpp/src/Ice/Properties.cpp:26]: It's just the implementation that was in PropertiesI - I didn't change anything.

**Review by pepone — APPROVED** [review]

File: cpp/src/Ice/Properties.cpp [hunk 4]
```diff
@@ -2,8 +2,729 @@
...
-Ice::Properties::~Properties()
+#include <fstream>
+
+using namespace std;
+using namespace Ice;
+using namespace IceInternal;
+
+Ice::Properties::Properties(const Properties& p)
+{
+    lock_guard lock(const_cast<Properties&>(p)._mutex);
```
Comment (externl) [inline cpp/src/Ice/Properties.cpp:22]: Maybe we should just make the mutex mutable?

**Review by externl — APPROVED** [review]
Looks good!

File: cpp/src/Ice/Properties.cpp (hunk 4 above)
Comment (bernardnormier) [reply cpp/src/Ice/Properties.cpp:22]: Fixed as suggested.

**Pull request merged**